File: Examples/CSharp/ModifyingAndConvertingImages/SyncRootProperty.cs
// GIST-ID: 73e5438d32fc70cc636540f0ba50fb60
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class SyncRootProperty
    {
        public static void Run()
        {
            // To get proper output, please apply a valid Aspose.Imaging license.
            // You can purchase a full license or obtain a 30‑day temporary license from https://www.aspose.com/purchase/default.aspx.

            // ExStart: SyncRootProperty

            // Create an instance of the MemoryStream class.
            using (System.IO.MemoryStream memoryStream = new System.IO.MemoryStream())
            {
                // Create an instance of the StreamContainer class and assign the memory stream object.
                using (StreamContainer streamContainer = new StreamContainer(memoryStream))
                {
                    // Check if the access to the stream source is synchronized.
                    lock (streamContainer.SyncRoot)
                    {
                        // Do work.
                        // Now access to the source MemoryStream is synchronized.
                    }
                }
            }

            // ExEnd: SyncRootProperty
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/SyncRootProperty.vb
Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages

    Public Class SyncRootProperty

        Public Shared Sub Run()
            
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx."

            ' ExStart: SyncRootProperty

            ' Create an instance of Memory stream class.
            Using memoryStream As New System.IO.MemoryStream()
                ' Create an instance of Stream container class and assign memory stream object.
                Using streamContainer As New StreamContainer(memoryStream)
                    ' check if the access to the stream source is synchronized.
                    ' do work
                    ' now access to source MemoryStream is synchronized
                    SyncLock streamContainer.SyncRoot
                    End SyncLock
                End Using
            End Using

            ' ExEnd: SyncRootProperty


        End Sub

    End Class

End Namespace
